sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -796,6 +800,109 @@ sycl::kernel k_float = kb.ext_oneapi_get_kernel("bartmpl<float>");
 sycl::kernel k_int = kb.ext_oneapi_get_kernel("bartmpl<int>");
 ----
 
+=== New kernel bundle member functions for device globals
+
+This extensions adds the following new `kernel_bundle` member functions to let
+the host application interact with device globals defined in runtime-compiled
+code. Device globals are only supported for the `source_language::sycl`
+language.
+
+Device globals declared with the `device_image_scope` property can be used in
+the runtime-compiled device code, but cannot be accessed from the host. We plan
+to lift this limitation in a future version of the extension.
+
+[source,c++]
+----
+namespace sycl {
+
+template <bundle_state State>
+class kernel_bundle {
+  // Continued from "New kernel bundle member functions"
+
+  bool ext_oneapi_has_device_global(const std::string &name, const device &dev);
+  void *ext_oneapi_get_device_global_address(const std::string &name,
+                                             const device &dev);
+  size_t ext_oneapi_get_device_global_size(const std::string &name,
+                                           const device &dev);
+};
+
+} // namespace sycl
+----
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+bool ext_oneapi_has_device_global(const std::string &name, const device &dev)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ The value `true` only if
+
+* the kernel bundle was created from a bundle of state
+  `bundle_state::ext_oneapi_source` in the language `source_language::sycl`, and
+* it defines a device global whose name is `name` and which was declared without
+  the `device_image_scope` property, and
+* `dev` is contained by the context associated with this bundle.
+
+`name` must be a {cpp} identifier that is valid for referencing the device
+global at the bottom of the source code.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+void *ext_oneapi_get_device_global_address(const std::string &name,
+                                           const device &dev)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Effects:_ If device memory for `name` has not been allocated at the time of
+this call, it will be allocated and zero-initialized synchronously.
+
+_Returns:_ Returns a USM pointer to the device global `name`'s storage on device
+`dev`.
```

Comment:
We should say that this is a "device USM" pointer.  That way application's know they cannot directly access it from the host.

The formatting is messed up here because of the "'s".  I generally avoid pluralizing identifiers for stylistic reasons.  I'd rewrite the sentence like:

> _Returns:_ A device USM pointer to the storage for the device global `name` on device `dev`.